pkg/kubelet/kubelet.go

```diff
@@ -2444,31 +2455,75 @@ func (kl *Kubelet) cleanUpContainersInPod(podID types.UID, exitedContainerID str
 	}
 }
 
-// fastStatusUpdateOnce starts a loop that checks the internal node indexer cache for when a CIDR
-// is applied  and tries to update pod CIDR immediately. After pod CIDR is updated it fires off
-// a runtime update and a node status update. Function returns after one successful node status update.
+// fastStatusUpdateOnce starts a loop that checks if the current state of kubelet + container runtime
+// would be able to turn the node ready, and sync the ready state to the API server as soon as possible.
+// Function returns after the node status update after such event, or when the node is already ready.
 // Function is executed only during Kubelet start which improves latency to ready node by updating
-// pod CIDR, runtime status and node statuses ASAP.
+// kubelet state, runtime status and node statuses ASAP.
 func (kl *Kubelet) fastStatusUpdateOnce() {
+	defer func() {
+		kl.containerRuntimeReadyExpected = true
```

Comment:
this is accessed from `func (kl *Kubelet) updateRuntimeUp()` too, do we need to hold this lock 
```
	kl.updateRuntimeMux.Lock()
	defer kl.updateRuntimeMux.Unlock()
```

or make an atomic bool? atomic.Bool is only a 1.19 golang thing, maybe the mutex is simpler?

Reply:
adding mutex.